UnityGLTF/Assets/UnityGLTF/Scripts/GLTFSceneImporter.cs

```diff
@@ -1115,15 +1115,18 @@ protected IEnumerator ConstructUnityMesh(MeshConstructionData meshConstructionDa
             var meshAttributes = meshConstructionData.MeshAttributes;
             var vertexCount = primitive.Attributes[SemanticProperties.POSITION].Value.Count;
 
-            // todo optimize: There are multiple copies being performed to turn the buffer data into mesh data. Look into reducing them
-            UnityEngine.Mesh mesh = new UnityEngine.Mesh
-            {
-                vertices = primitive.Attributes.ContainsKey(SemanticProperties.POSITION)
-                    ? meshAttributes[SemanticProperties.POSITION].AccessorContent.AsVertices.ToUnityVector3Raw()
-                    : null,
-                normals = primitive.Attributes.ContainsKey(SemanticProperties.NORMAL)
-                    ? meshAttributes[SemanticProperties.NORMAL].AccessorContent.AsNormals.ToUnityVector3Raw()
-                    : null,
+			// todo optimize: There are multiple copies being performed to turn the buffer data into mesh data. Look into reducing them
+			UnityEngine.Mesh mesh = new UnityEngine.Mesh
+			{
+#if UNITY_2017_3_OR_NEWER
+				indexFormat = vertexCount > 65535 ? IndexFormat.UInt32 : IndexFormat.UInt16,

```

Comment:
Can you have a case where you have less than 65535 indices that are being stored as 32 bit integers?

Reply:
Even if they were stored in 32 bit values you wouldn't lose anything by putting them in to shorts in the mesh.